### ISSUE TITLE
Studio attempts to load shared libraries from /usr

### ISSUE THREAD

**Issue by eugeneia** [studio/studio#48]
When I press “play” I get thrown into the Debugger because Studio attempted to load a shared library from `/usr` instead of `/nix`. Below are the Debugger stack trace and the console output:

```
ExternalLibraryFunction(Object)>>error:
ExternalLibraryFunction(Object)>>externalCallFailed
ExternalLibraryFunction(ExternalFunction)>>invokeWithArguments:
AthensCairoSurface class>>primImage:width:height:
FFICalloutAPI>>function:module:
AthensCairoSurface class(Object)>>nbCall:
AthensCairoSurface class>>primImage:width:height:
AthensCairoSurface class>>extent:format:
AthensCairoSurface class>>extent:
TRCanvas>>initialize
TRCanvas class(Behavior)>>new
RTView>>initialize
RTView class(Behavior)>>new
RTMondrian(RTBuilder)>>createView
RTMondrian>>createView
RTMondrian(RTBuilder)>>initialize
RTMondrian>>initialize
RTMondrian class(Behavior)>>new
RJITAuditLog>>visualizeMondrianOverview
RJITProcess>>gtInspectorJITIn:
[ :eachPragma | 
eachPragma methodSelector numArgs = 0
	ifTrue: [ | presentationSource |
		presentationSource := self perform: eachPragma selector.
		presentationSource glmPresentation
			cull: composite
			cull: aGTInspector
			cull: self ].
eachPragma methodSelector numArgs = 1
	ifTrue: [ thisContext
			object: self
			perform: eachPragma methodSelector
			withArguments: {composite}
			inClass: self class ].
eachPragma methodSelector numArgs = 2
	ifTrue: [ thisContext
			object: self
			perform: eachPragma methodSelector
			withArguments:
				{composite.
				aGTInspector}
			inClass: self class ] ] in RJITProcess(ProtoObject)>>gtInspectorPresentationsFromPragmas:In:inContext:
Array(SequenceableCollection)>>do:
RJITProcess(ProtoObject)>>gtInspectorPresentationsFromPragmas:In:inContext:
RJITProcess(ProtoObject)>>gtInspectorPresentationsIn:inContext:
[ :d | each gtInspectorPresentationsIn: d inContext: self ] in [ :x | 
(GLMCompositePresentation new
	with:
		[ :d | each gtInspectorPresentationsIn: d inContext: self ])
	startOn: x ] in [ :a :each | 
a title: [ self printObjectAsAnItem: each ].
a
	dynamicActions: [ :aPresentation | each gtInspectorActions asOrderedCollection ].
a titleAction: [ :aPresentation | self closeActionFor: aPresentation ].
a dynamic
	display: [ :x | 
		(GLMCompositePresentation new
			with:
				[ :d | each gtInspectorPresentationsIn: d inContext: self ])
			startOn: x ] ] in [ :browser | 
browser fixedSizePanes: self class numberOfInspectorPanes.
browser
	show: [ :a :each | 
		a title: [ self printObjectAsAnItem: each ].
		a
			dynamicActions: [ :aPresentation | each gtInspectorActions asOrderedCollection ].
		a
			titleAction: [ :aPresentation | self closeActionFor: aPresentation ].
		a dynamic
			display: [ :x | 
				(GLMCompositePresentation new
					with:
						[ :d | each gtInspectorPresentationsIn: d inContext: self ])
					startOn: x ] ] ] in StudioInspector(GTInspector)>>compose
GLMCompositePresentation(GLMPresentation)>>with:
[ :x | 
(GLMCompositePresentation new
	with:
		[ :d | each gtInspectorPresentationsIn: d inContext: self ])
	startOn: x ] in [ :a :each | 
a title: [ self printObjectAsAnItem: each ].
a
	dynamicActions: [ :aPresentation | each gtInspectorActions asOrderedCollection ].
a titleAction: [ :aPresentation | self closeActionFor: aPresentation ].
a dynamic
	display: [ :x | 
		(GLMCompositePresentation new
			with:
				[ :d | each gtInspectorPresentationsIn: d inContext: self ])
			startOn: x ] ] in [ :browser | 
browser fixedSizePanes: self class numberOfInspectorPanes.
browser
	show: [ :a :each | 
		a title: [ self printObjectAsAnItem: each ].
		a
			dynamicActions: [ :aPresentation | each gtInspectorActions asOrderedCollection ].
		a
			titleAction: [ :aPresentation | self closeActionFor: aPresentation ].
		a dynamic
			display: [ :x | 
				(GLMCompositePresentation new
					with:
						[ :d | each gtInspectorPresentationsIn: d inContext: self ])
					startOn: x ] ] ] in StudioInspector(GTInspector)>>compose
BlockClosure>>glamourValueWithArgs:
BlockClosure(ProtoObject)>>glamourValue:
GLMDynamicPresentation(GLMPresentation)>>displayValue
GLMDynamicPresentation>>currentPresentation
GLMMorphicDynamicRenderer>>render:
GLMMorphicDynamicRenderer class(GLMMorphicWidgetRenderer class)>>render:from:
GLMMorphicRenderer>>renderDynamicPresentation:
GLMDynamicPresentation>>renderGlamorouslyOn:
GLMMorphicRenderer(GLMRenderer)>>render:
GLMMorphicTabbedRenderer(GLMMorphicWidgetRenderer)>>renderObject:
GLMMorphicTabbedRenderer(GLMMorphicWidgetRenderer)>>renderWithTitleOrNil:
GLMMorphicTabbedRenderer>>render:
GLMMorphicTabbedRenderer class(GLMMorphicWidgetRenderer class)>>render:from:
GLMMorphicRenderer>>renderTabbedCompositePresentation:
GLMTabbedArrangement>>renderGlamorouslyOn:
GLMCompositePresentation>>renderGlamorouslyOn:
GLMMorphicRenderer(GLMRenderer)>>render:
GLMMorphicPaneRenderer(GLMMorphicWidgetRenderer)>>renderObject:
GLMMorphicPaneRenderer>>renderContainerFor:
GLMMorphicPaneRenderer>>render:
GLMMorphicPaneRenderer class(GLMMorphicWidgetRenderer class)>>render:from:
GLMMorphicRenderer>>renderPane:
GLMPane>>renderGlamorouslyOn:
GLMMorphicRenderer(GLMRenderer)>>render:
GLMMorphicPagerRenderer(GLMMorphicWidgetRenderer)>>renderObject:
GLMMorphicPagerRenderer>>actOnPaneAdded:
MessageSend>>value:
MessageSend>>cull:
MessageSend>>cull:cull:
[ action cull: anAnnouncement cull: announcer ] in AnnouncementSubscription>>deliver:
BlockClosure>>on:do:
BlockClosure>>on:fork:
AnnouncementSubscription>>deliver:
[ "Ensure delivery to remaining announcements" subscription deliver: anAnnouncement ] in SubscriptionRegistry>>deliver:to:startingAt:
BlockClosure>>ifCurtailed:
SubscriptionRegistry>>deliver:to:startingAt:
SubscriptionRegistry>>deliver:to:
SubscriptionRegistry>>deliver:
GLMPager(Announcer)>>announce:
GLMPager(GLMAnnouncer)>>announce:
GLMPager(GLMLoggedObject)>>announce:
GLMPager(GLMFinder)>>innerPortEvent:
GLMPane>>portEvent:
[ self silentValue: anObject.
self pane
	portEvent:
		(GLMPortEvent
			on: self
			previouslyValued: oldValue
			in: aTransmissionContext) ] in GLMPanePort>>changeValueTo:in:
GLMPane>>notingPresentationChangeDo:
GLMPanePort>>changeValueTo:in:
GLMPanePort>>value:in:
GLMPanePort(GLMPort)>>value:
GLMPortUpdater>>glamourValueWithArgs:
GLMGenericAction(GLMAction)>>actOn:
GLMGenericAction(GLMAction)>>morphicActOn:
GLMPluggableButtonMorph(PluggableButtonMorph)>>performAction:
[ :m | 
(m containsPoint: evt cursorPoint)
	ifTrue: [ m enabled
			ifTrue: [ m performAction: evt ] ] ] in GLMPluggableButtonMorph(PluggableButtonMorph)>>mouseUp:
Array(SequenceableCollection)>>do:
GLMPluggableButtonMorph(PluggableButtonMorph)>>mouseUp:
GLMPluggableButtonMorph(Morph)>>handleMouseUp:
MouseButtonEvent>>sentTo:
GLMPluggableButtonMorph(Morph)>>handleEvent:
GLMPluggableButtonMorph(Morph)>>handleFocusEvent:
[ ActiveHand := self.
ActiveEvent := anEvent.
result := focusHolder
	handleFocusEvent: (anEvent transformedBy: (focusHolder transformedFrom: self)) ] in HandMorph>>sendFocusEvent:to:clear:
BlockClosure>>on:do:
WorldMorph(PasteUpMorph)>>becomeActiveDuring:
HandMorph>>sendFocusEvent:to:clear:
HandMorph>>sendEvent:focus:clear:
HandMorph>>sendMouseEvent:
HandMorph>>handleEvent:
HandMorph>>processEventsFromQueue:
HandMorph>>processEvents
[ :h | 
self activeHand: h.
h processEvents.
self activeHand: nil ] in WorldState>>doOneCycleNowFor:
Array(SequenceableCollection)>>do:
WorldState>>handsDo:
WorldState>>doOneCycleNowFor:
WorldState>>doOneCycleFor:
WorldMorph>>doOneCycle
WorldMorph class>>doOneCycle
[ [ WorldMorph doOneCycle.
Processor yield.
false ] whileFalse: [  ] ] in MorphicUIManager>>spawnNewProcess
[ self value.
Processor terminateActive ] in BlockClosure>>newProcess
```

```
% studio-x11
setlocale() failed (check values of LC_CTYPE, LANG and LC_ALL)
pthread_setschedparam failed: Operation not permitted
ioLoadModule(//usr/lib/libcairo.so.2):
  libpixman-1.so.0: cannot open shared object file: No such file or directory
Error: External module not found
ExternalLibraryFunction(Object)>>error:
ExternalLibraryFunction(Object)>>externalCallFailed
ExternalLibraryFunction(ExternalFunction)>>invokeWithArguments:
AthensCairoSurface class>>primImage:width:height:
FFICalloutAPI>>function:module:
AthensCairoSurface class(Object)>>nbCall:
AthensCairoSurface class>>primImage:width:height:
AthensCairoSurface class>>extent:format:
AthensCairoSurface class>>extent:
TRCanvas>>initialize
TRCanvas class(Behavior)>>new
RTView>>initialize
RTView class(Behavior)>>new
RTMondrian(RTBuilder)>>createView
RTMondrian>>createView
RTMondrian(RTBuilder)>>initialize
RTMondrian>>initialize
RTMondrian class(Behavior)>>new
RJITAuditLog>>visualizeMondrianOverview
RJITProcess>>gtInspectorJITIn:
[ :eachPragma | 
eachPragma methodSelector numArgs = 0
	ifTrue: [ | presentationSource |
		presentationSource := self perform: eachPragma selector.
		presentationSource glmPresentation
			cull: composite
			cull: aGTInspector
			cull: self ].
eachPragma methodSelector numArgs = 1
	ifTrue: [ thisContext
			object: self
			perform: eachPragma methodSelector
			withArguments: {composite}
			inClass: self class ].
eachPragma methodSelector numArgs = 2
	ifTrue: [ thisContext
			object: self
			perform: eachPragma methodSelector
			withArguments:
				{composite.
				aGTInspector}
			inClass: self class ] ] in RJITProcess(ProtoObject)>>gtInspectorPresentationsFromPragmas:In:inContext: in Block: [ :eachPragma | ...
Array(SequenceableCollection)>>do:
RJITProcess(ProtoObject)>>gtInspectorPresentationsFromPragmas:In:inContext:
RJITProcess(ProtoObject)>>gtInspectorPresentationsIn:inContext:
[ :d | each gtInspectorPresentationsIn: d inContext: self ] in [ :x | 
(GLMCompositePresentation new
	with:
		[ :d | each gtInspectorPresentationsIn: d inContext: self ])
	startOn: x ] in [ :a :each | 
a title: [ self printObjectAsAnItem: each ].
a
	dynamicActions: [ :aPresentation | each gtInspectorActions asOrderedCollection ].
a titleAction: [ :aPresentation | self closeActionFor: aPresentation ].
a dynamic
	display: [ :x | 
		(GLMCompositePresentation new
			with:
				[ :d | each gtInspectorPresentationsIn: d inContext: self ])
			startOn: x ] ] in [ :browser | 
browser fixedSizePanes: self class numberOfInspectorPanes.
browser
	show: [ :a :each | 
		a title: [ self printObjectAsAnItem: each ].
		a
			dynamicActions: [ :aPresentation | each gtInspectorActions asOrderedCollection ].
		a
			titleAction: [ :aPresentation | self closeActionFor: aPresentation ].
		a dynamic
			display: [ :x | 
				(GLMCompositePresentation new
					with:
						[ :d | each gtInspectorPresentationsIn: d inContext: self ])
					startOn: x ] ] ] in StudioInspector(GTInspector)>>compose in Block: [ :d | each gtInspectorPresentationsIn: d inContex...etc...
GLMCompositePresentation(GLMPresentation)>>with:
[ :x | 
(GLMCompositePresentation new
	with:
		[ :d | each gtInspectorPresentationsIn: d inContext: self ])
	startOn: x ] in [ :a :each | 
a title: [ self printObjectAsAnItem: each ].
a
	dynamicActions: [ :aPresentation | each gtInspectorActions asOrderedCollection ].
a titleAction: [ :aPresentation | self closeActionFor: aPresentation ].
a dynamic
	display: [ :x | 
		(GLMCompositePresentation new
			with:
				[ :d | each gtInspectorPresentationsIn: d inContext: self ])
			startOn: x ] ] in [ :browser | 
browser fixedSizePanes: self class numberOfInspectorPanes.
browser
	show: [ :a :each | 
		a title: [ self printObjectAsAnItem: each ].
		a
			dynamicActions: [ :aPresentation | each gtInspectorActions asOrderedCollection ].
		a
			titleAction: [ :aPresentation | self closeActionFor: aPresentation ].
		a dynamic
			display: [ :x | 
				(GLMCompositePresentation new
					with:
						[ :d | each gtInspectorPresentationsIn: d inContext: self ])
					startOn: x ] ] ] in StudioInspector(GTInspector)>>compose in Block: [ :x | ...
BlockClosure>>glamourValueWithArgs:
BlockClosure(ProtoObject)>>glamourValue:
GLMDynamicPresentation(GLMPresentation)>>displayValue
```

**Comment by lukego** [comment]
@eugeneia Could you please check if this is resolved by #61?

I had the idea that we could run the whole Pharo GUI inside a chroot where it would not be able to see directories like `/usr` but I am not sure if that is straightforward. So I patched the Cairo binding to stop looking in hard-coded paths instead.

**Comment by eugeneia** [comment]
Nope, sorry. I get the exact same error as before.
